src/Factory/LoggerFactory.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Factory;
+
+use Psr\Container\ContainerInterface;
+use Psr\Log\LoggerInterface;
+use RuntimeException;
+use Yiisoft\Factory\Definitions\DefinitionInterface;
+use Yiisoft\Factory\Definitions\Normalizer;
+use Yiisoft\Factory\Factory;
+
+final class LoggerFactory extends Factory
+{
+    private static ?ContainerInterface $container = null;
+    private static ?DefinitionInterface $definition = null;
+    private static Factory $loggerFactory;
+
+    private function __construct(ContainerInterface $container = null)
+    {
+        self::$container = $container;
+        self::$definition = Normalizer::normalize(LoggerInterface::class);
+    }
+
+    public static function initialize(ContainerInterface $container = null): void
+    {
+        self::$loggerFactory = new self($container);
+    }
+
+    /**
+     * Creates a `LoggerInterface` instance.
+     *
+     * @throws RuntimeException If the created object is not an instance of the `LoggerInterface`.
+     *
+     * @return LoggerInterface The `LoggerInterface` instance.
+     *
+     * @psalm-suppress RedundantConditionGivenDocblockType
+     * @psalm-suppress DocblockTypeContradiction
+     */
+    public static function run(): LoggerInterface
```

Comment:
```suggestion
    public static function create(): LoggerInterface
```

Reply:
Won't this create new instance each time? If not, that's not really a factory because factory is expected to create new instance each time.

Reply:
Can remove inheritance, as it is redundant here. Then you can rename the method from `run()` to `create()`.

Reply:
done.

Reply:
done.